src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,106 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    static final int GROWTH_FACTOR = 2;
+    private int size;
+    private Node<K, V>[] mapArray;
+
+    public MyHashMap() {
+        mapArray = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node newNode = new Node<>(key, value, null);
+        insert(index, newNode, key, value);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> oldNode = mapArray[getIndex(key)];
+        if (oldNode == null) {
+            return null;
+        }
+        while (true) {
```

Comment:
[endless loop? ](https://stackoverflow.com/questions/390481/is-while-true-with-break-bad-programming-practice)
use condition here.